### PR TITLE
Keyboard shortcut to copy password

### PR DESCRIPTION
**Pull request by respawned86**
Sometimes you want to copy a password in order to paste it into an application where Bitwarden cannot autofill it.

Previously, you had to use your mouse or tab through a lot of buttons and submenus.

Now, you can go to your browser, activate Bitwarden's popup, search for the password, and copy the password using only your keyboard.

### CODE REVIEW

**Comment by cscharf** [comment]
Thank you @respawned86 for the PR! We will review this in the next couple of weeks (we have a lot of folks out on vacation at the moment).

**Comment by BryceBarbara** [comment]
@cscharf Any update on the review status of this?

**Comment by cscharf** [comment]
Hi @BryceBarbara , no, this is on our contributions board to get reviewed, unfortunately we've have some shifts in process and this got lost along the way. @respawned86 , it appears there are no some merge conflicts (due to the passing of time), my apologies but if you could resolve those, I have this on the top of our review swimlane on our contributions board for after we cut our release candidate branch tomorrow/Monday.

**Review by eliykat — CHANGES_REQUESTED** [review]
Hi @respawned86, sorry for the delay on the review! I've looked over your code and I think it's an excellent addition. I think using directives like this is a good solution.

Aside from the comment below, my only request is that we follow the same [hotkeys as the desktop application](https://bitwarden.com/help/article/keyboard-shortcuts/):
* Ctrl/CMD + U → Copy Username
* Ctrl/CMD + P → Copy Password
* Ctrl/CMD + T → Copy TOTP

I also suggest the following additional hotkeys for complete coverage of the action buttons:

* Ctrl/CMD + I → view **I**tem (mostly useful from the Tab page where Enter will autofill)
* Ctrl/CMD + O → **O**pen URL

That way, we can accomplish anything using hotkeys that we can using the GUI. But let me know if you have different ideas or feedback about the above.

I'll keep an eye out for your response so that we can get this merged :)

File: src/directives/copy-paste.directive.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @HostListener('keydown', ['$event']) onKeyDown($event: KeyboardEvent) {
+        this.getEventEmitter($event)?.emit();
+    }
+
+    private getEventEmitter($event: KeyboardEvent) {
+        if (!$event.ctrlKey || $event.shiftKey || $event.altKey || $event.metaKey || $event.repeat)
+            return null;
+
+        switch ($event.code) {
+            case 'KeyX':
```
Comment (eliykat) [inline src/directives/copy-paste.directive.ts:25]: Is there any use case for cut and paste? If not I'd rather remove them until we actually need them.

File: src/directives/copy-paste.directive.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @Output() cut = new EventEmitter();
+    @Output() copy = new EventEmitter();
+    @Output() paste = new EventEmitter();
+
+    @HostListener('keydown', ['$event']) onKeyDown($event: KeyboardEvent) {
+        this.getEventEmitter($event)?.emit();
+    }
+
+    private getEventEmitter($event: KeyboardEvent) {
+        if (!$event.ctrlKey || $event.shiftKey || $event.altKey || $event.metaKey || $event.repeat)
```
Comment (eliykat) [inline src/directives/copy-paste.directive.ts:21]: Style: please use curly braces for all blocks regardless of whether they are 1 line or multiple lines

**Review by eliykat — CHANGES_REQUESTED** [review]

**Comment by matschaffer** [comment]
@eliykat I'm wondering if we should close this one for similar reasons listed in https://github.com/bitwarden/browser/pull/2492#issuecomment-1090995717

Both PRs were called out to me from https://community.bitwarden.com/t/add-essential-keyboard-shortcuts-navigation/76/137 but I agree that we probably need a new PR before we can consider merging.

**Comment by eliykat** [comment]
Hey @matschaffer sorry I missed your message until now. I think this PR is a little different in that the technical approach is basically sound, it just needs some tweaking in the implementation. This is a case where you can fix merge conflicts, address my feedback above, and I think it would be ready for merging. I can still close it though if you're going to submit a fresh PR, @respawned86 isn't going to progress it, or it ages further.

**Comment by eliykat** [comment]
I'm closing this as there's been no further movement on this PR.

**Pull request closed without merging**